### PR TITLE
py-obspy: Add py312 subport

### PR DESCRIPTION
**Pull request by jolivetr**
…e obspy312.

#### Description

Update obspy portfile to python 3.12

###### Type(s)

- [ ] bugfix
- [X] enhancement
- [ ] security fix

###### Tested on
macOS 12.7.6 21H1320 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [X] checked your Portfile with `port lint --nitpick`?
- [X] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by jolivetr** [comment]
Building fails when attempting to build with py38 and py39. I suggest we remove support for these old versions of python.

Who makes that decision? Obspy developers, Portfile owner or someone else?

File: python/py-obspy/Portfile [hunk 1]
```diff
@@ -26,7 +26,7 @@ checksums           rmd160  3e861220a8616ba1e68f16851296c197e3d80b90 \
                     sha256  336a6e1d9a485732b08173cb5dc1dd720a8e53f3b54c180a62bb8ceaa5fe5c06 \
                     size    16984057
 
-python.versions     38 39 310 311
+python.versions     38 39 310 311 312
 python.pep517       yes
```
Comment (reneeotten) [inline python/py-obspy/Portfile:30]: this is the default now and the line can thus be removed

**Review by reneeotten — CHANGES_REQUESTED** [review]
there is one test dependency missing for the PY312 subport which the linter warns you about:
```
port lint py312-obspy:
Error: Unknown dependency: py312-pytest-json-report
``` 

that should be resolved in this PR as well. 

Additionally, your commit message does not follow the guidelines - it should be ```portname: short description```. So in this case something like: ```py-obspy: add py312 subport```.

**Comment by reneeotten** [comment]
> Building fails when attempting to build with py38 and py39. I suggest we remove support for these old versions of python.
> 
> Who makes that decision? Obspy developers, Portfile owner or someone else?

Upstream claims to support Python 3.8 and above so technically these subports should build.... I will look at the CI logs in a minute. Having said that, this port has no dependents and it's reasonable to assume people can use a modern Python version. Typically, the policy is to remove subports for EOL versions (which PY38 almost is) and at this time removing PY39 would be fine as well; a port maintainer can decide which subports to add.

**Comment by reneeotten** [comment]
the PY38/PY39 subports fail because one of their dependencies doesn't install:
```
  Building dependency (1 of 1) 'py38-cartopy' with variants '' ... [FAIL] (checksum)
```

that is something that should be fixed; ideally in this PR as well. 

Having said that, if you'd like to remove the almost EOL Python 3.8 subport that's fine. It would be preferable though to fix the ```checksums``` issue regardless.

**Comment by ryandesign** [comment]
> ```
> Error: Unknown dependency: py312-pytest-json-report
> ```

This is fixed by b0819b8d5fa85e7c8971ed74a26e3885f9ad4a3c and dd7800d8ff0e3139dbbd05d3625a365cd96a81f1

> ```
>   Building dependency (1 of 1) 'py38-cartopy' with variants '' ... [FAIL] (checksum)
> ```

This is fixed by 20a29fa1c6e0abd3432ccd2664f298a61015a6d6.

> your commit message does not follow the guidelines

I've rebased this PR onto the new master and reworded the commit message and force-pushed.

Romain, for future pull requests you'll want to create a dedicated branch for your changes rather than making them on master.

**Comment by ryandesign** [comment]
Romain, since your master differs from ours, you'll want to either rebase your master from ours and force-push, or delete your clone and (if you want to make more contributions) recreate it.

**Pull request merged**